src/sentry/runner/commands/backup.py

```diff
@@ -16,9 +16,22 @@
 def import_(src):
     "Imports data from a Sentry export."
 
-    for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
-        if obj.object._meta.app_label not in EXCLUDED_APPS:
-            obj.save()
+    try:
+        with transaction.atomic():
+            for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
+                if obj.object._meta.app_label not in EXCLUDED_APPS:
+                    obj.save()
+    # For all database integrity errors, let's warn users to follow our
+    # recommended backup/restore workflow before reraising exception. Most of
+    # these errors come from restoring on a different version of Sentry or not restoring
+    # on a clean install.
+    except IntegrityError as e:
+        warningText = ">> Are you restoring from a backup of the same version of Sentry?\n>> Are you restoring onto a clean database?\n>> If you're doing both already, file a bug report, it's probably our fault"
```

Comment:
Worth putting a link in there?

```
https://github.com/getsentry/sentry/issues/new/choose
```

Reply:
I mean, I think we should either put in the link, or leave out the "If you're doing both ..." line.

Reply:
Hmmm, it's probably better to tell people where to file an issue instead of making them look for it 🙈

changed!